src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/ChainedMethodCallWrapCheck.java

```diff
@@ -0,0 +1,448 @@
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.checks.whitespace;
+
+import java.util.regex.Pattern;
+
+import com.puppycrawl.tools.checkstyle.StatelessCheck;
+import com.puppycrawl.tools.checkstyle.api.AbstractCheck;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.FullIdent;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+import com.puppycrawl.tools.checkstyle.utils.TokenUtil;
+
+/**
+ * <p>
+ * Checks whether method calls specified by identifier pattern are wrapped.
+ * Wrapping means displaying the contents of the current line on the next line.
+ * It is as simple as adding a line break on the current line and continuing
+ * it on the next line.
+ * </p>
+ * <p>
+ * Rationale: Long chained method calls are difficult to read and comprehend and sometimes
+ * require horizontal scrolling to read the complete method call chain. Wrapping them onto
+ * the next line/s enhances readability.
+ * </p>
+ * <ul>
+ * <li>
+ * Property {@code identifierPattern} - Specify method calls to be checked.
+ * By default, no method calls are checked.
+ * Type is {@code java.util.regex.Pattern}.
+ * Default value is {@code "^$"}.
+ * </li>
+ * <li>
+ * Property {@code maxCallsPerLine} - Specify maximum number of method calls per lines if we
+ * are wrapping the method calls. If all the method calls span over a single line then this
+ * property has no effect.
+ * Type is {@code int}.
+ * Default value is {@code 1}.
+ * </li>
+ * <li>
+ * Property {@code maxSingleLineCalls} - Specify maximum number of method calls if all chained
+ * method calls span over a single line. If chained method calls span over multiple lines then
+ * this property has no effect.
+ * Type is {@code int}.
+ * Default value is {@code 1}.
+ * </li>
+ * </ul>
+ * <p>
+ * To configure the check to use custom identifierPattern to validate assertions:
+ * </p>
+ * <pre>
+ * &lt;module name=&quot;ChainedMethodCallWrap&quot;&gt;
+ *  &lt;property name=&quot;identifierPattern&quot; value=&quot;^assert.*$&quot; /&gt;
+ * &lt;/module&gt;
+ * </pre>
+ * <p>
+ * Example:
+ * </p>
+ * <pre>
+ * void method() {
+ *     assertWithSomeMessage("ex").that(1).isEqualTo(1);
+ *                      // violation above, ''3' method calls on single line, max allowed is '1''
+ *     assertWithSomeMessage("ex")
+ *         .that(1).isEqualTo(1); // violation, 'Chained method call should be wrapped'
+ *
+ *     assertWithSomeMessage("ex")
```

Comment:
Just curious, why `assertWithSomeMessage` and not `assertWithMessage`?

Reply:
I had to do this because of-
https://github.com/checkstyle/checkstyle/blob/c20fc8141aaee3f041d7dd583e5db256ac0f5a99/config/checkstyle_checks.xml#L135-L141
```
Truth's 'that' method call should be on separate line [assertThatShouldBeOnSeparateLine]
```

Reply:
Can we modify regexp for this module in a supplemental commit, to not violate lines that begin with `*` ?

Reply:
Done as a minor commit

Reply:
Resolved.